r/src/type_infer.cpp

```diff
@@ -165,8 +165,13 @@ std::shared_ptr<arrow::DataType> InferArrowTypeFromVector<VECSXP>(SEXP x) {
         cpp11::stop(
             "Requires at least one element to infer the values' type of a list vector");
       }
-
-      ptype = VECTOR_ELT(x, 0);
+      // Iterate through the vector until we get a non-null result
+      for (R_xlen_t i = 0; i < XLENGTH(x); i++) {
+        ptype = VECTOR_ELT(x, i);
+        if (!Rf_isNull(ptype)) {
```

Comment:
What happens if all elements are NULL?

Reply:
We get the error `Cannot infer type from vector`.

Reply:
I guess we actually want the Null type there...

Reply:
Updated now